ambry-cloud/src/main/java/com/github/ambry/cloud/VcrReplicationManager.java

```diff
@@ -183,6 +186,28 @@ protected ReplicaThread getReplicaThread(String threadName, FindTokenHelper find
         replicaSyncUpManager, skipPredicate, leaderBasedReplicationAdmin, this.persistor, cloudDestination, properties);
   }
 
+  /**
+   * Assign all replicas of a partition to the same thread to avoid concurrent updates to Azure
+   * @param remoteReplicaInfos List of {@link RemoteReplicaInfo}
+   * @param startThread start threads on creation
+   */
+  @Override
+  protected void addRemoteReplicaInfoToReplicaThread(List<RemoteReplicaInfo> remoteReplicaInfos, boolean startThread) {
+    for (RemoteReplicaInfo rinfo : remoteReplicaInfos) {
+      String dc = rinfo.getReplicaId().getDataNodeId().getDatacenterName();
+      List<ReplicaThread> replicaThreads = getOrCreateThreadPoolIfNecessary(dc, startThread);
+      if (replicaThreads != null) {
+        ReplicaThread rthread = partitionToReplicaThread.computeIfAbsent(
+            rinfo.getLocalReplicaId().getPartitionId().getId(),
+            key -> replicaThreads.get(getReplicaThreadIndexToUse(dc)));
+        rthread.addRemoteReplicaInfo(rinfo);
+        rinfo.setReplicaThread(rthread);
```

Comment:
This may cause each thread to talk to more Ambry servers? For example, if ReplicaThread1 is responsible for P1, P2, P3, it could talk to 9 hosts (in worst case) in each replication cycle. Just wondering if it could cause increase in replication lag. I think tests may help to know if there are any side effects.

Reply:
hmm good point, lemme think abt it

Reply:
A single thread already talks to multiple nodes today. There are more number of nodes than there are threads.